### PR TITLE
Add sending of contact message to GQL API

### PR DESCRIPTION
**Pull request by cfranklin11**
Currently, submission of contact forms are sent as emails to the site admin. I don't know what format these emails have, so I made up a reasonable subject & body.

Mocking external modules in TypeScript is kind of a pain, so I wasn't able to write a working test for SendGrid's error handling, because the correctly-typed error response didn't have the form that SendGrid actually returns, so I just gave up.

### CODE REVIEW

File: src/graphql/index.ts [hunk 1]
```diff
@@ -121,6 +124,52 @@ async function loadSchema(connection: DbConnection): Promise<GraphQLSchema> {
...
+      status: {
+        type: messageStatusEnum
+      },
+      message: {
+        type: GraphQLNonNull(GraphQLString)
+      }
+    })
+  })
+
+  async function resolveSendContactMessage(root, args, context): Promise<SendEmailResponse> {
```
Comment (meligatt) [inline src/graphql/index.ts:147]: I see you set the context in another file like this: `const context = { sendEmail: Email.send }`

Is this `context` arg a graphql thing or you can pass whatever you want as an arg to that resolver? (after `root` and `args`)

Comment (meligatt) [reply src/graphql/index.ts:147]: nevermind ..I just saw that ApolloServer() expects that stuff

File: src/email.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import sgMail from '@sendgrid/mail'
+
+import type { SendEmailResponse } from './types'
+
+type SendOptions = {
```
Comment (meligatt) [inline src/email.ts:5]: I see that you have a `./types` in src folder, should  this sendOptions be here in this email.ts file of in that types one? ... or you add types in src/types when you need to reuse them?

File: test/email.spec.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    const sendOptions = {
+      to: faker.internet.email(),
+      from: faker.internet.email(),
+      subject: faker.company.bs(),
+      text: faker.lorem.paragraph()
+    }
+
+    mockedSgMail.send.mockReturnValue(Promise.resolve([new Response(200, {}), {}]))
+
+    it("calls sendgrid's send function", async () => {
```
Comment (meligatt) [inline test/email.spec.ts:23]: Do you need this test case ? isn't just enough the one below, it("returns a result object") ..I understand why you expect the call of `sendgrid's send` but isn't that internal implementation? (that is what sometimes confuses me still)

File: src/email.ts (hunk 2 above)
Comment (cfranklin11) [reply src/email.ts:5]: I use it in two different places, so I put it in the general `types` module. When I use a type in one place, I just keep it in the same module where I use it.

File: test/email.spec.ts (hunk 3 above)
Comment (cfranklin11) [reply test/email.spec.ts:23]: This function has two effects:
1. The side-effect of sending an email
2. Returning a response object

It's generally a best practice to break up expectations of what a unit does into separate test functions/blocks (unless exercising a particular class/function is very slow), which is why I have two separate `it` calls.

Checking that SendGrid's `send` function gets called is necessary to make sure that we're trying to send an email. It's equivalent to testing a function that creates a DB record by querying the DB for the record that was created. Just checking the response wouldn't actually test whether we tried to send an email: if you comment out the call to SendGrid's `send` function, the test for the response will still pass.

Comment (meligatt) [reply test/email.spec.ts:23]: got it

**Pull request merged**